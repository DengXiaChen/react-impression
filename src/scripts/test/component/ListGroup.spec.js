import React from 'react';
import { shallow } from 'enzyme';
import { ListGroup } from '../../components/impression';

const setup = () => {
    const props = [{
        href: '/listGroup',
        disabled: true,
    }, {
        active: true,
    }];

    const wrapper = shallow(
        <ListGroup>
            <ListGroup.Item {...props[0]}>ListGroup1</ListGroup.Item>
            <ListGroup.Item {...props[1]}>ListGroup2</ListGroup.Item>
        </ListGroup>
    );

    return {
        props,
        wrapper,
    };
};

describe('ListGroup', () => {
    const { wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('ul.list-group').exists()).toBe(true);
    });

    it('href', () => {
        expect(wrapper.render().find('a.list-group-item').prop('href')).toBe('/listGroup');
    });

    it('disabled', () => {
        expect(wrapper.render().find('.list-group-item.disabled')).toHaveLength(1);
    });

    it('header', () => {
        expect(wrapper.render().find('.list-group-item.active')).toHaveLength(1);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
