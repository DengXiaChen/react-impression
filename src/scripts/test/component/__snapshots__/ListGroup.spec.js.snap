// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="list-group"
>
    <ListGroupItem
        disabled={true}
        href="/listGroup"
    >
        ListGroup1
    </ListGroupItem>
    <ListGroupItem
        active={true}
    >
        ListGroup2
    </ListGroupItem>
</ul>,
  "nodes": Array [
    <ul
      className="list-group"
>
      <ListGroupItem
            disabled={true}
            href="/listGroup"
      >
            ListGroup1
      </ListGroupItem>
      <ListGroupItem
            active={true}
      >
            ListGroup2
      </ListGroupItem>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ListGroup>
        <ListGroupItem
                disabled={true}
                href="/listGroup"
        >
                ListGroup1
        </ListGroupItem>
        <ListGroupItem
                active={true}
        >
                ListGroup2
        </ListGroupItem>
</ListGroup>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <ListGroupItem
              disabled={true}
              href="/listGroup"
>
              ListGroup1
</ListGroupItem>,
            <ListGroupItem
              active={true}
>
              ListGroup2
</ListGroupItem>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="list-group"
>
          <ListGroupItem
                    disabled={true}
                    href="/listGroup"
          >
                    ListGroup1
          </ListGroupItem>
          <ListGroupItem
                    active={true}
          >
                    ListGroup2
          </ListGroupItem>
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="list-group"
>
          <ListGroupItem
                    disabled={true}
                    href="/listGroup"
          >
                    ListGroup1
          </ListGroupItem>
          <ListGroupItem
                    active={true}
          >
                    ListGroup2
          </ListGroupItem>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ListGroup>
    <ListGroupItem
        disabled={true}
        href="/listGroup"
    >
        ListGroup1
    </ListGroupItem>
    <ListGroupItem
        active={true}
    >
        ListGroup2
    </ListGroupItem>
</ListGroup>,
}
`;
